### PR TITLE
Create firejail-welcome.s

### LINKED ISSUE #3797: Get ride of all these u2f and drm issues
I'm becoming quite annoyed by all those issues about u2f and drm in browsers or other common and easy to solve issues.

**Proposal**

We should provide a setup-script for new users. Example: https://gist.github.com/rusty-snake/17494a3ad9366ce46a6b2bcc4c46149b.

Usage:
```
wget -O- https://gist.githubusercontent.com/rusty-snake/17494a3ad9366ce46a6b2bcc4c46149b/raw/da671d6fe92b38179904f9bb8428eab030088859/firejail-welcome.sh | sh
```

_well you know `wget | sh` is :bomb: :stopwatch: :boom: ._

Dependencies:

bash, sed, coreutils, sudo, zenity

### PR DESCRIPTION
**Pull request by rusty-snake**
- fix #3797 -- Get ride of all these u2f and drm issues
 - @netblue30 https://github.com/netblue30/firejail/issues/3797#issuecomment-743110173 ("add it to https://firejail.wordpress.com/documentation-2/ >> I’ve just installed Firejail, now what?!")
 -  @ everyone: Improve the messages/descriptions to make them more newbie friendly and more detailed.

### CODE REVIEW

File: contrib/firejail-welcome.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+export LANG=en_US.UTF8
+
+zenity --title=firejail-welcome.sh --text-info --width=750 --height=500 <<EOM
+Welcome to firejail!
+
+This is a quick setup guide for newbies.
+
+Profiles for programs can be found in /etc/firejail. Own customizations should go in a file named
+<proile-name>.local in ~/.config/firejal.
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:15]: typo: profile-name

File: contrib/firejail-welcome.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+Firejails own configuration can be found at /etc/firejail/firejail.config.
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:17]: Firejail's

File: contrib/firejail-welcome.sh [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+Please note that running this script a second time can set new options, but does not unset options
+set in a previous run.
+
+Webiste: https://firejail.wordpress.com
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:22]: Website

File: contrib/firejail-welcome.sh [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+Bug-Tracker: https://github.com/netblue30/firejail/issues
+Documentation:
+- https://github.com/netblue30/firejail/wiki
+- https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions
+- https://firejail.wordpress.com/documentation-2
+- man:firejail(1) and man:firejail-profile(5)
+
+PS: If you have any improvements for this script, open a issues or pull request.
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:30]: an issue

File: contrib/firejail-welcome.sh [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+sed_scripts=()
+
+read -r -d $'\0' MSG_Q_BROWSER_DISABLE_U2F <<EOM
+<big><b>Should browsers be allowed to access u2f hardware?</b></big>
+EOM
+
+read -r -d $'\0' MSG_Q_BROWSER_ALLOW_DRM <<EOM
+<big><b>Should browsers be able to play DRM content?</b></big>
+
+\$HOME is noexec,nodev,nosuid by default for the most sandboxes. This means that executing ELFs which are located in \$HOME,
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:43]: do newbies know what ELFs means? maybe use "programs" or "application binaries".

File: contrib/firejail-welcome.sh [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+is forbidden, the setuid attribute on files is ignored and device files inside \$HOME don't work. Browsers install proprietary
+DRM plug-ins such as Widevine under \$HOME by default. In order to use them, \$HOME must be mounted exec inside the sandbox to
+allow there execution. Clearly, this may help an attacker to start malicious code.
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:46]: their

File: contrib/firejail-welcome.sh [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+NOTE: Other software written in an interpreter language such as bash, python or java can always started from \$HOME.
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:48]: can always *be* started.

File: contrib/firejail-welcome.sh [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+TIPP: If <tt>/home</tt> has its own partition, you can mount it <tt>nodev,nosuid</tt> for all programs.
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:50]: Tipp is German. Use "Tip" or "Hint". :-)

File: contrib/firejail-welcome.sh [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+EOM
+
+read -r -d $'\0' MSG_L_ADVANCED_OPTIONS <<EOM
+You maybe want to set some of these advanced options.
+EOM
+
+read -r -d $'\0' MSG_Q_RUN_FIRECFG <<EOM
+<big><b>Should the most programs started in firejail by default?</b></big>
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:58]: Should most programs be/get started ...

File: contrib/firejail-welcome.sh [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+EOM
+
+read -r -d $'\0' MSG_I_ROOT_REQUIRED <<EOM
+In order to apply these changes, root right are required.
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:62]: maybe use "root privileges"?

File: contrib/firejail-welcome.sh [hunk 11]
```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+# This file is part of Firejail project
+# Copyright (C) 2014-2020 Firejail Authors
+# License GPL v2
+
+export LANG=en_US.UTF8
+
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:8]: zenity and sudo might not be installed. consider checking for them at the beginning and abort (and print an error message) if they are not available.

File: contrib/firejail-welcome.sh [hunk 12]
```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+# This file is part of Firejail project
+# Copyright (C) 2014-2020 Firejail Authors
```
Comment (reinerh) [inline contrib/firejail-welcome.sh:4]: as thils file was written in 2020, the correct copyright would probably be just 2020.

**Comment by reinerh** [comment]
I like this script! it's guiding beginners easily through a few common setup steps.

**Review by reinerh — APPROVED** [review]

**Comment by netblue30** [comment]
Cool! I would merge it in firecfg. If you run as regular user "firecfg --welcome" it should start the script.

The problem: this will bring in a new dependency, for example in Debian it would bring in zenity, gtk, webkitgtk and so on. 

An idea would be to detect if zenity is installed, and if not use something similar to zenity that works in console.

**Pull request merged**